clang/runtime/dpct-rt/include/dpct/codepin/serialization/serial_basic.hpp

```diff
@@ -0,0 +1,155 @@
+//==---- serial_basic.hpp ----------------------------*-C++-*-------------==//
+//
+// Copyright (C) Intel Corporation
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===--------------------------------------------------------------------===//
+#ifndef __DPCT_CODEPIN_SERIAL_BASIC_HPP__
+#define __DPCT_CODEPIN_SERIAL_BASIC_HPP__
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#ifdef __NVCC__
+#include <cuda_runtime.h>
+#else
+#include <dpct/dpct.hpp>
+#include <sycl/sycl.hpp>
+#endif
+
+namespace dpct {
+namespace experimental {
+
+#ifdef __NVCC__
+typedef cudaStream_t StreamType;
+#else
+typedef dpct::queue_ptr StreamType;
+#endif
+
+namespace detail {
+
+template <class T, class T2 = void> class TT {
+public:
+  static void dump(std::ostream &ss, T value,
+                   dpct::experimental::StreamType stream) {
+    ss << "{\"Type\":\"" << typeid(T).name() << "\",\"Data\":[";
+    ss << "Unable to find the corresponding serialization function for the "
+          "class. Please report this issue to SYCLomatic/DPCT, or manually add "
+          "the serialization function.";
```

Comment:
try to define the error code for L38,  1. as the logfile is expected to be quite big, not readable for most fields.  2.make compare script easy.